frontend/src/pages/ChainEdit.js

```diff
@@ -1,85 +1,198 @@
-import { TextFormField } from "../components/FormFields";
 import GeocoderSelector from "../components/GeocoderSelector";
+import { TextForm, CheckboxField } from "../components/FormFields";
+import ThreeColumnLayout from "../components/ThreeColumnLayout";
 
 const ChainEdit = () => {
   const { t } = useTranslation();
-  const {chainId} = useParams()
+  let history = useHistory();
+  let location = useLocation();
+  const { chainId } = useParams();
+
   const [chain, setChain] = useState();
   const classes = makeStyles(theme)();
-  const { register, handleSubmit, setValue } = useForm();
-  const history = useHistory();
   const [address, setAddress] = useState();
+  const [coordinates, setCoordinates] = useState();
+  const [submitted, setSubmitted] = useState(false);
+  const [error, setError] = useState("");
+  const [categoriesState, setCategoriesState] = useState({});
+
+  const validate = Yup.object({
+    name: Yup.string().min(2, "Must be more than 2 characters"),
+    description: Yup.string().min(2, "Must be more than 2 characters"),
+  });
+
+  const handleChange = (e) => {
+    if (e.target.checked) {
+      setCategoriesState({
+        ...categoriesState,
+        [e.target.name]: true,
+      });
+    } else {
+      setCategoriesState({
+        ...categoriesState,
+        [e.target.name]: false,
+      });
+    }
+  };
 
-  const onSubmit = (formData) => {
-    let chain = {
-      ...formData,
-      address: address,
+  //refactor db data from array to obj
+  const mapCat = (array) => {
+    return {
+      men: array.includes("men"),
+      woman: array.includes("woman"),
+      mix: array.includes("mix"),
     };
-    updateChain(chainId, chain);
-    history.push({
-      pathname: `/chains/members/${chainId}`,
-      state: { message: t("saved") },
-    });
-    console.log("submitted");
   };
 
   useEffect(async () => {
     const chain = await getChain(chainId);
     setChain(chain);
+    setAddress(chain.address);
+    setCoordinates({
+      latitude: chain.latitude,
+      longitude: chain.longitude,
+    });
+
+    setCategoriesState(mapCat(chain.categories.gender));
   }, []);
 
-  return !chain ? null : <>
-    <Helmet>
-      <title>Clothing-chain | Edit chain details</title>
-      <meta name="description" content="Edit chain details" />
-    </Helmet>
-    <Grid container className={classes.form}>
-      <Grid item sm />
-      <Grid item sm>
-        <Typography variant="h3" className={classes.pageTitle}>
-          {t("edit chain details")}
-        </Typography>
+  return !chain ? null : (
+    <>
+      <Helmet>
+        <title>Clothing-chain | Edit chain details</title>
+        <meta name="description" content="Edit chain details" />
+      </Helmet>{" "}
+      <Formik
+        initialValues={{
+          name: chain.name,
+          description: chain.description,
+        }}
+        validationSchema={validate}
+        onSubmit={async (values) => {
```

Comment:
Probably best to move the submit logic to the code. Keep some separation between logic and layout code.